File: geoservice-mobile/lib/models/auth_response.dart
/// Modèle pour la réponse d'authentification JWT
class AuthResponse {
  final String accessToken;
  final String refreshToken;
  final String tokenType;
  final int expiresIn;
  final int userId;
  final String username;
  final String email;
  final String fullName;
  final String role;

  AuthResponse({
    required this.accessToken,
    required this.refreshToken,
    required this.tokenType,
    required this.expiresIn,
    required this.userId,
    required this.username,
    required this.email,
    required this.fullName,
    required this.role,
  });

  /// Crée une instance à partir d'une réponse JSON
  factory AuthResponse.fromJson(Map<String, dynamic> json) {
    return AuthResponse(
      accessToken: json['accessToken'] ?? '',
      refreshToken: json['refreshToken'] ?? '',
      tokenType: json['tokenType'] ?? 'Bearer',
      expiresIn: json['expiresIn'] ?? 3600,
      userId: json['userId'] ?? 0,
      username: json['username'] ?? '',
      email: json['email'] ?? '',
      fullName: json['fullName'] ?? '',
      role: json['role'] ?? 'USER',
    );
  }

  /// Convertit l'instance en JSON
  Map<String, dynamic> toJson() {
    return {
      'accessToken': accessToken,
      'refreshToken': refreshToken,
      'tokenType': tokenType,
      'expiresIn': expiresIn,
      'userId': userId,
      'username': username,
      'email': email,
      'fullName': fullName,
      'role': role,
    };
  }
}

File: geoservice-isochrone/src/main/java/com/github/cunvoas/geoserviceisochrone/controller/mobile/auth/dto/AuthResponse.java
package com.github.cunvoas.geoserviceisochrone.controller.mobile.auth.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * DTO pour la réponse d'authentification.
 * Contient les tokens JWT et les informations de l'utilisateur.
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AuthResponse {

    private String accessToken;
    private String refreshToken;
    private String tokenType = "Bearer";
    private Long expiresIn;
    
    // Informations de l'utilisateur
    private Long userId;
    private String username;
    private String email;
    private String fullName;
    private String role;
}
